gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -230,6 +232,29 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
     {% endif %}
     {% endfor %}
 
+{# TODO: Allow client creation to explictly specify connection
+         parameters, such as secure/insecure connections, and have tests use
+         that. Then remove this function.
+         Tracking issue:
+         https://github.com/googleapis/gapic-generator-python/issues/901
+#}
+def _protocol_for(host: str) -> str:
+    """Determine the URL protocol to use for connecting to the given host.
+
+    This function returns 'http' for hosts that match
+    `^localhost(:.*)?$` and `https` for all others. This is intended
+    to allow integration testing over an insecure connection to a
+    locally running server.
+
+    Args:
+        host (str): The hostname for REST connection.
+
+    Returns:
+        str: The protocol (`http` or `https`) to use for connecting to `host`.
+    """
+    if not hasattr(_protocol_for, 'regex'):
+        _protocol_for.regex = re.compile("^localhost(:.*)?$")
```

Comment:
Will this regex be executed for every call made? If yes, it will have some performance implications (not big probably, but here we sacrifice production code to support rare testing case, which is suboptimal).

Reply:
The compilation of the regex will only happen once because I have it memoized via this attribute.

The matching will happen for every call for now, though we could refactor things so the protocol is only computed once per client and stored in an attribute. However, given that the performance hit is small, I'm inclined to think it's preferable to keep things simple.